src/main/java/com/powsybl/openloadflow/network/impl/Networks.java

```diff
@@ -222,9 +224,21 @@ public static LfNetworkList load(Network network, LfNetworkParameters networkPar
             List<LfNetwork> lfNetworks = load(network, topoConfig, networkParameters, reportNode);
 
             if (!(modifiedTopoConfig.getSwitchesToClose().isEmpty() && modifiedTopoConfig.getBranchIdsToClose().isEmpty())) {
+                Set<String> switchAndBranchIdsLeftToClose = modifiedTopoConfig.getSwitchesToClose().stream()
+                        .filter(Objects::nonNull)
+                        .map(Identifiable::getId)
+                        .collect(Collectors.toSet());
+                switchAndBranchIdsLeftToClose.addAll(modifiedTopoConfig.getBranchIdsToClose());
                 for (LfNetwork lfNetwork : lfNetworks) {
+                    // all switches and branches were closed
+                    if (switchAndBranchIdsLeftToClose.isEmpty()) {
+                        break;
+                    }
                     // disable all buses and branches not connected to main component (because of switch to close)
-                    restoreInitialTopology(lfNetwork, modifiedTopoConfig.getSwitchesToClose(), modifiedTopoConfig.getBranchIdsToClose());
+                    restoreInitialTopology(lfNetwork, switchAndBranchIdsLeftToClose);
+                }
+                if (!switchAndBranchIdsLeftToClose.isEmpty()) {
```

Comment:
I don't understand why we could be in that case (with the exception). If we are in that case, it means that we have a bug in the code. Any advice @geofjamg ?

Reply:
Either:
  - we are able to add a test case for this case because it is a real case
  - we remove the code because it cannot happen

Reply:
It is not a real case, I just put it to be sure that we would catch a bug if it happened. I will remove this bit of code.